Nasm.md

```diff
@@ -252,3 +252,68 @@ je .value3_case
 ```
 
 ## Data structures
+
+As you can imagine assembly doesn't have intrinsic concept of data structures, but luckily we have the support of the preprocessor where that let us declare struct like blocks.
+
+This guide will just introduce quickly how to define a basic struct,for more information and use cases is better to check the netwide assembler official documentation (see the useful links section)
+
+Let's for example assume we have the following C struct:
+
+```c
+struct task {
+    uint32_t id;
+    char name[8];
+};
+```
+
+How nasm render a struct is basically declaring a list of offset labels in this way  we can use them to access the field starting from the struct memory location (*Authors note: yeah it is a trick...*)
+To create a struct in nasm we use the `struc` and `endstruc` keywords, and the fields are defined between these them. 
+The example above can be rendered in the following way:
+
+```asm
+struc task
+    id:         resd    1
+    name:       resd 8
+endstruc
+```
+
+what this code is doing is creating three symbols: id as 0 representing the offset from the beginning of a task structure and name as 4 (still the offset) and the task symbol that is 0 too. This notation has a drawback, it defines the labels as global constants, so you can't have another struct or label declared with same name, to solve this problem you can use the following notation: 
+
+```asm
+struc task
+    .id:    resd    1
+    .name:  resb    8
+endstruc
+```
+
+In this case the ofssets label created are `task.id`, `task.name`. So to put it in simpler terms what does we really have here are some label that represents offset, and to access the fields of a structure what we do is adding these offsets to the memory location containing it.
```

Comment:
This whole line can be simplified a bit I think: 
Now we can access the fields inside our struct in a familiar way: `struct_name.field_name`. What's really happening here is the assembler will add the offset of `field_name` to the base address of `struct_name` to give us the real address of this variable.